pkgs/applications/emulators/box64/default.nix

```diff
@@ -5,77 +5,95 @@
-      hash = "sha256-11hy5Ol5FSE/kNJmXAIwNLbapldhlZGKtOLIoL6pYrg=";
-    })
-  ];
-
-  nativeBuildInputs = [
-    cmake
-    python3
-  ];
-
-  cmakeFlags = [
-    "-DNOGIT=1"
-  ] ++ (
-    if stdenv.hostPlatform.system == "aarch64-linux" then
-      [
-        "-DARM_DYNAREC=ON"
-      ]
-    else [
-      "-DLD80BITS=1"
-      "-DNOALIGN=1"
-    ]
-  );
-
-  installPhase = ''
-    runHook preInstall
-    install -Dm 0755 box64 "$out/bin/box64"
-    runHook postInstall
-  '';
-
-  doCheck = true;
-
-  checkPhase = ''
-    runHook preCheck
-    ctest
-    runHook postCheck
-  '';
-
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    runHook preInstallCheck
-    $out/bin/box64 -v
-    runHook postInstallCheck
-  '';
-
-  passthru.updateScript = gitUpdater {
-    rev-prefix = "v";
-  };
+    src = fetchFromGitHub {
+      owner = "ptitSeb";
+      repo = pname;
+      rev = "v${version}";
+      hash = "sha256-eMp2eSWMRJQvLRQKUirBua6Kt7ZtyebfUnKIlibkNFU=";
+    };
+
+    patches = [
+      # Fix mmx & cppThreads tests on x86_64
+      # Remove when version > 0.2.0
+      (fetchpatch {
+        url = "https://github.com/ptitSeb/box64/commit/3819aecf078fcf47b2bc73713531361406a51895.patch";
+        hash = "sha256-11hy5Ol5FSE/kNJmXAIwNLbapldhlZGKtOLIoL6pYrg=";
+      })
+    ];
+
+    nativeBuildInputs = [
+      cmake
+      python3
+    ];
+
+    cmakeFlags = [
+      "-DNOGIT=ON"
+      "-DARM_DYNAREC=${if withDynarec then "ON" else "OFF"}"
+      "-DRV64=${if stdenv.hostPlatform.isRiscV64 then "ON" else "OFF"}"
+      "-DPPC64LE=${if stdenv.hostPlatform.isPower64 && stdenv.hostPlatform.isLittleEndian then "ON" else "OFF"}"
+    ] ++ lib.optionals stdenv.hostPlatform.isx86_64 [
+      "-DLD80BITS=ON"
+      "-DNOALIGN=ON"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm 0755 box64 "$out/bin/box64"
+
+      runHook postInstall
+    '';
+
+    doCheck = true;
+
+    doInstallCheck = true;
+
+    installCheckPhase = ''
+      runHook preInstallCheck
+
+      echo Checking if it works
+      $out/bin/box64 -v
+
+      echo Checking if Dynarec option was respected
+      $out/bin/box64 -v | grep ${lib.optionalString (!withDynarec) "-v"} Dynarec
+
+      runHook postInstallCheck
+    '';
+
+    passthru = {
+      updateScript = gitUpdater {
+        rev-prefix = "v";
+      };
+    };
 
-  meta = with lib; {
-    homepage = "https://box86.org/";
-    description = "Lets you run x86_64 Linux programs on non-x86_64 Linux systems";
-    license = licenses.mit;
-    maintainers = with maintainers; [ gador ];
-    platforms = [ "x86_64-linux" "aarch64-linux" ];
+    meta = with lib; {
+      homepage = "https://box86.org/";
+      description = "Lets you run x86_64 Linux programs on non-x86_64 Linux systems";
+      license = licenses.mit;
+      maintainers = with maintainers; [ gador OPNA2608 ];
+      platforms = [ "x86_64-linux" "aarch64-linux" "riscv64-linux" "powerpc64le-linux" ];
+    };
   };
-}
+in
```

Comment:
We don't need to hide the package inside a let in. We can add box64 to inputs and then use it in passthru.

Reply:
I moved it into the `let` because the `box64` from the inputs is not the `box64` that's being built when you do `box64.override { withDynarec = false; }`, which is kinda annoying for testing if the variant still works.

Reply:
Meh, fine…